src/Datadog.Trace/Agent/AgentWriter.cs

```diff
@@ -2,30 +2,41 @@
 using System.Collections.Generic;
 using System.Linq;
 using System.Threading.Tasks;
+using Datadog.Trace.DogStatsD;
 using Datadog.Trace.Logging;
+using Datadog.Trace.Vendors.StatsdClient;
 
 namespace Datadog.Trace.Agent
 {
     internal class AgentWriter : IAgentWriter
     {
+        private const int TraceBufferSize = 1000;
+
         private static readonly Vendors.Serilog.ILogger Log = DatadogLogging.For<AgentWriter>();
 
-        private readonly AgentWriterBuffer<List<Span>> _tracesBuffer = new AgentWriterBuffer<List<Span>>(1000);
+        private readonly AgentWriterBuffer<List<Span>> _tracesBuffer = new AgentWriterBuffer<List<Span>>(TraceBufferSize);
         private readonly IApi _api;
+        private readonly IDogStatsd _dogStatsdClient;
         private readonly Task _flushTask;
         private readonly TaskCompletionSource<bool> _processExit = new TaskCompletionSource<bool>();
 
-        public AgentWriter(IApi api)
+        public AgentWriter(IApi api, IDogStatsd dogStatsdClient)
         {
             _api = api;
+            _dogStatsdClient = dogStatsdClient;
             _flushTask = Task.Run(FlushTracesTaskLoopAsync);
         }
 
         public void WriteTrace(List<Span> trace)
         {
             var success = _tracesBuffer.Push(trace);
+
+            _dogStatsdClient?.Increment(TracerMetricNames.Queue.EnqueuedTraces);
+            _dogStatsdClient?.Increment(TracerMetricNames.Queue.EnqueuedSpans, trace.Count);
```

Comment:
I see that dogstatsd uses UDP, but is there a way in there to batch these count metrics?

Reply:
It looks like `Increment` uses `Send` underneath, which will immediately send the commands up.
If we are able to use `Add` and *then* call `Send`, then we can batch any related stats in one request.

Reply:
This is great, thanks!

Reply:
Switched from `IDogStatsd` to `IStatsd` and used `Add()` and `Send()` to batch commands.